### PR TITLE
Parametrize environment to use

### PR DESCRIPTION
**Pull request by mcardosos**
@seyadava PTAL

Thanks for your contribution! Please let us know what problems this PR addresses or what functionality it adds. Please link to other related issues and PRs to speed up review. Thank you!

### CODE REVIEW

File: authorization/auth_test.go [hunk 1]
```diff
@@ -21,6 +22,11 @@ func TestMain(m *testing.M) {
 		log.Fatalln("failed to parse args")
 	}
 
+	err = iam.ParseArgs()
```
Comment (jhendrixMSFT) [inline authorization/auth_test.go]: How come we have two ParseArgs() funcs to call?

File: helpers/parse_args.go [hunk 2]
```diff
@@ -12,18 +12,23 @@ import (
 	"os"
 	"strings"
 
+	"github.com/Azure/go-autorest/autorest/azure"
 	"github.com/subosito/gotenv"
 )
 
+const (
+	publicCloudName = "AZUREPUBLICCLOUD"
```
Comment (jhendrixMSFT) [inline helpers/parse_args.go]: Is this const something that should be defined in go-autorest?

File: helpers/parse_args.go [hunk 3]
```diff
@@ -83,12 +88,17 @@ func ParseArgs() error {
 	if os.Getenv("AZURE_SAMPLES_KEEP_RESOURCES") == "1" {
 		keepResources = true
 	}
-	armEndpointString = os.Getenv("AZURE_ARM_ENDPOINT")
+
+	envName = os.Getenv("AZURE_ENVIRONMENT")
```
Comment (jhendrixMSFT) [inline helpers/parse_args.go]: I'm thinking that we should define these well-known environment vars as exported constants in go-autorest.  What do you think?

Comment (mcardosos) [reply helpers/parse_args.go]: I would agree

File: authorization/auth_test.go (hunk 1 above)
Comment (mcardosos) [reply authorization/auth_test.go]: One is for authentication, the other is for environments

File: helpers/parse_args.go (hunk 3 above)
Comment (marstr) [reply helpers/parse_args.go]: I think we should go a step further and create a [viper](https://github.com/spf13/viper) instance in go-autorest that looks for them in a number of places, including environment variables. 🐍

File: helpers/parse_args.go (hunk 2 above)
Comment (marstr) [reply helpers/parse_args.go]: It sort of is already: https://github.com/Azure/go-autorest/blob/9ad9326b278af8fa5cc67c30c0ce9a58cc0862b2/autorest/azure/environments.go#L63

File: authorization/auth_test.go (hunk 1 above)
Comment (mcardosos) [reply authorization/auth_test.go]: Fixed

File: helpers/parse_args.go [hunk 4]
```diff
@@ -18,12 +18,8 @@ import (
 	"github.com/subosito/gotenv"
 )
 
-const (
-	publicCloudName = "AZUREPUBLICCLOUD"
-)
-
 var (
-	envName                  = publicCloudName
+	tangetEnv                = azure.PublicCloud.Name
```
Comment (jhendrixMSFT) [inline helpers/parse_args.go]: targetEnv?

Comment (mcardosos) [reply helpers/parse_args.go]: Typo, lol. Thanks for pointing at this

**Pull request merged**